File: leetcode3/변지협/v2/1539. Kth Missing Positive Number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        i = 1
        lmt = 0
        arr = arr[::-1]
        while True:
            if len(arr) == 0:
                lmt += 1
            elif arr[-1] == i:
                arr.pop()
            else:
                lmt += 1
            
            if lmt == k:
                break
            i += 1

        return i

File: leetcode3/황은지/1539. Kth Missing Positive Number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    const set=new Set(arr);

    let count=0;

    for(let i=1;i<=100000;i++){
        if(!set.has(i)){
            count++;
            if(count===k) return i;
        }
    }
};